### PR TITLE
[BEAM-886] Remove the usage of OldDoFn and clean up function names

### PR DESCRIPTION
**Pull request by sb2nov**
Be sure to do all of the following to help us incorporate your contribution
quickly and easily:

 - [x] Make sure the PR title is formatted like:
   `[BEAM-<Jira issue #>] Description of pull request`
 - [x] Make sure tests pass via `mvn clean verify`. (Even better, enable
       Travis-CI on your fork and ensure the whole test matrix passes).
 - [x] Replace `<Jira issue #>` in the title with the actual Jira issue
       number, if there is one.
 - [ ] If this contribution is large, please file an Apache
       [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.txt).

---

R: @aaltay PTAL

### CODE REVIEW

File: sdks/python/apache_beam/runners/common.pxd [hunk 1]
```diff
@@ -37,22 +37,20 @@ cdef class DoFnRunner(Receiver):
   cdef object tagged_receivers
   cdef LoggingContext logging_context
   cdef object step_name
-  cdef bint is_new_dofn
   cdef list args
   cdef dict kwargs
   cdef ScopedMetricsContainer scoped_metrics_container
   cdef list side_inputs
-  cdef bint has_windowed_side_inputs
+  cdef bint windowed_inputs
```
Comment (aaltay) [inline sdks/python/apache_beam/runners/common.pxd]: Name this `has_windowed_inputs` maybe, because it still a boolean and not a list of inputs.

Comment (sb2nov) [inline sdks/python/apache_beam/runners/common.pxd]: Done.

**Comment by sb2nov** [comment]
Thanks. @aaltay PTAL

**Comment by sb2nov** [comment]
retest this please

**Comment by aaltay** [comment]
LGTM. Could you re-run the tests? Let's make sure at least one of them is green.

**Pull request merged**